kms/attestations/verify_attestation.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+
+# Copyright 2020 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This application verifies HSM attestations using certificate bundles
+obtained from Cloud HSM.
+
+For more information, visit https://cloud.google.com/kms/docs/attest-key.
+"""
+
+# [START verify_attestations]
+import argparse
+import gzip
+
+from cryptography import exceptions
+from cryptography import x509
+from cryptography.hazmat import backends
+from cryptography.hazmat.primitives.asymmetric import padding
+import pem
+
+
+def get_uncompressed_contents(filename):
+    """Gets the uncompressed contents of a file.
+
+    Args:
+        filename: The name of the file.
+
+    Returns:
+        The uncompressed contents of the file.
+    """
+    try:
+        with gzip.open(filename, 'rb') as f:
+            return f.read()
+    except IOError:
+        with open(filename, 'rb') as f:
+            return f.read()
+
+
+def verify(attestation_file, bundle_file):
+    """Verifies an attestation using a bundle of certificates.
+
+    Args:
+      attestation_file: The name of the attestation file.
+      bundle_file: The name of the bundle file containing the certificates
+        used to verify the attestation.
+
+    Returns:
+      True if at least one of the certificates in bundle_file can verify the
+      attestation statement and its signature.
+    """
+    attestation = get_uncompressed_contents(attestation_file)
+    statement = attestation[:-256]
+    signature = attestation[-256:]
```

Comment:
What's the significance of 256? Will every file be formatted this way?

Reply:
Yes, I'll add a comment.